### PR TITLE
Re-order extension methods

### PR DESCRIPTION
**Pull request by Corniel**
Re-ordered extension methods:
* all in `Extensions` folder
* Extensions on Type in file with name reflecting name and namespace
* Extension name `$"Qowaiv{Type.Name}Extensions"`
* Namespace equal to namespace of Type to extend on
* Extensions on Enumerables in `System.Linq` namespace

Humanizer extensions
* Namespace of type to create

### CODE REVIEW

File: specs/Qowaiv.Specs/Extensions/Enumerable_specs.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using FluentAssertions;
+using NUnit.Framework;
+using Qowaiv.Statistics;
+using System.Linq;
+
+namespace Extensions.Enumerable_specs
+{
+    public class Avarage
```
Comment (LauraXiulan) [inline specs/Qowaiv.Specs/Extensions/Enumerable_specs.cs:8]: Average

File: specs/Qowaiv.Specs/Extensions/Type_specs.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+{
+    public class CSharpString
+    {
+        [TestCase(typeof(void), "void")]
+        [TestCase(typeof(byte), "byte")]
+        [TestCase(typeof(float), "float")]
+        [TestCase(typeof(double), "double")]
+        [TestCase(typeof(object), "object")]
+        public void Uses_primitive_names(Type primitive, string csharpString)
+            => Assert.That(primitive.ToCSharpString(), Is.EqualTo(csharpString));
```
Comment (LauraXiulan) [inline specs/Qowaiv.Specs/Extensions/Type_specs.cs:16]: Inconsistent use of Assert and Should

File: specs/Qowaiv.Specs/IO/StreamSize_specs.cs [hunk 3]
```diff
@@ -1,14 +1,108 @@
...
 using Qowaiv.Specs;
 using Qowaiv.TestTools;
 using Qowaiv.TestTools.Globalization;
+using Qowaiv.TestTools.IO;
+using System.Globalization;
+using System.IO;
 
 namespace IO.StreamSize_specs
 {
+    public class Has_humanzizer_creators
```
Comment (LauraXiulan) [inline specs/Qowaiv.Specs/IO/StreamSize_specs.cs:14]: humanizer

File: specs/Qowaiv.Specs/Mathematics/Fraction_specs.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+using Qowaiv.Mathematics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Mathematics.Fraction_specs
+{
+    public class Has_humanzizer_creators
```
Comment (LauraXiulan) [inline specs/Qowaiv.Specs/Mathematics/Fraction_specs.cs:12]: humanizer

File: src/Qowaiv/DecimalRound.cs [hunk 5]
```diff
@@ -4,81 +4,11 @@
...
-        /// The factor of which the number should be multiple of.
-        /// </param>
-        /// <param name="mode">
-        /// The rounding method used to determine the closed by number.
-        /// </param>
-        /// <returns>
-        /// A rounded number that is multiple to the specified factor.
-        /// </returns>
-        [Pure]
-        public static decimal RoundToMultiple(this decimal value, decimal multipleOf, DecimalRounding mode)
```
Comment (LauraXiulan) [inline src/Qowaiv/DecimalRound.cs:49]: RoundToMultiple not needed anymore?

File: src/Qowaiv/Extensions/System.Linq.Enumerable.cs [hunk 6]
```diff
@@ -418,6 +411,16 @@ public static Money Average(this IEnumerable<Money> source)
             return (total / count) + currency;
         }
 
+        /// <summary>Gets the average Elo.</summary>
+        [Pure]
+        public static Elo Avarage(this IEnumerable<Elo> elos) => elos.Select(elo => (double)elo).Average();
```
Comment (LauraXiulan) [inline src/Qowaiv/Extensions/System.Linq.Enumerable.cs:416]: Average

File: src/Qowaiv/Extensions/System.Linq.Enumerable.cs [hunk 7]
```diff
@@ -510,11 +512,8 @@ public static Amount Sum<TSource>(this IEnumerable<TSource> source, Func<TSource
                     }
                 }
             }
-            if (none)
-            {
-                return null;
-            }
-            return (Amount)total;
+            if (none) return null;
```
Comment (LauraXiulan) [inline src/Qowaiv/Extensions/System.Linq.Enumerable.cs:515]: return none ? null : total.Amount();

File: src/Qowaiv/Extensions/System.Linq.Enumerable.cs [hunk 8]
```diff
@@ -590,11 +587,8 @@ public static Amount Sum(this IEnumerable<Amount> source)
...
```
Comment (LauraXiulan) [inline src/Qowaiv/Extensions/System.Linq.Enumerable.cs:590]: return none ? null : total.Amount();

File: src/Qowaiv/DecimalRound.cs (hunk 5 above)
Comment (Corniel) [reply src/Qowaiv/DecimalRound.cs:49]: Exposed by extensions. So only the root method is needed that is exposed via extensions too.

File: src/Qowaiv/Extensions/System.Linq.Enumerable.cs (hunk 6 above)
Comment (Corniel) [reply src/Qowaiv/Extensions/System.Linq.Enumerable.cs]: Oops, oops, oops.

**Pull request merged**